src/sentry/utils/services.py

```diff
@@ -228,6 +234,12 @@ def __getattr__(self, attribute_name):
             return base_value
 
         def execute(*args, **kwargs):
+            # If this thread already has an active backend for this base class,
+            # we can safely call that backend synchronously without delegating.
+            if self in self.state.backends:
+                backend = type(self).state.backends[self.__backend_base]
```

Comment:
Also for posterity, this `type(self)` can actually be `self` but I thought it was better to make it explicit that this is a class attribute and not an instance attribute so that it can be accessed externally.

Reply:
Haha, I was going to ask, but then I got what it was doing. Figured some pro :snake: was going down here.